### PR TITLE
#36124 Truncate History entries

### LINKED ISSUE #36124: Text of the item in a list of a history is very long
### Description

- use the query for dvdrental database:
```
select * from public.rental as r
join inventory as i on i.inventory_id = r.inventory_id
join public.customer as c on c.customer_id = r.customer_id
join staff as st on st.staff_id = r.staff_id
join film as f on f.film_id = i.film_id
join address as a1 on a1.address_id = c.address_id
join address as a2 on a2.address_id = st.address_id
join "language" as l on f.language_id = l.language_id
;
```

https://github.com/user-attachments/assets/cefa52da-620b-439f-a259-2dd2fdbfa986

### DBeaver Version

Enterprise Edition 24.2.4-202410172322

### Operating System

Windows 10

### Database and driver

PostgreSQL 16
PostgreSQL JDBC Driver 42.7.4

### Steps to reproduce

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by IvanGorshechnikov**
image

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java [hunk 1]
```diff
@@ -5112,6 +5112,9 @@ String describeState() {
                 SQLUtils.appendConditionString(filter, context.getDataSource(), null, condBuffer, true);
                 desc += " [" + condBuffer + "]";
             }
+            if (desc != null && desc.length() > 50) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java]: Let's make constant for 50

**Review by serge-rider — CHANGES_REQUESTED** [review]

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java (hunk 1 above)
Comment (IvanGorshechnikov) [reply plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java]: Fix it

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java [hunk 2]
```diff
@@ -142,10 +142,11 @@ public class ResultSetViewer extends Viewer
     private static final String TOOLBAR_EXPORT_CONTRIBUTION_ID = "toolbar:org.jkiss.dbeaver.ui.controls.resultset.status.exportCmd";
     private static final String TOOLBAR_CONTRIBUTION_ID = "toolbar:org.jkiss.dbeaver.ui.controls.resultset.status";
     private static final String TOOLBAR_CONFIGURATION_VISIBLE_PROPERTY = "org.jkiss.dbeaver.ui.toolbar.configuration.visible";
-    
+
     private static final String CONFIRM_SERVER_SIDE_ORDERING_UNAVAILABLE = "org.jkiss.dbeaver.sql.resultset.serverSideOrderingUnavailable";
 
     private static final int THEME_UPDATE_DELAY_MS = 250;
+    private static final int MAXIMAL_LENGTH_HISTORY_STATE_ITEM = 50;
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetViewer.java]: Naming convention is ZZZ_MAX_LENGTH.
It is also better to declare constant in HistoryStateItem , it is not used anywhere else

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified

**Pull request merged**